Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift

```diff
@@ -14,6 +15,98 @@ protocol PitchSpellingNode: Hashable {
     var index: Index { get }
 }
 
+struct Reducer<S, A>: Monoid {
+    static var identity: Reducer {
+        return .init { _, _ in return }
+    }
+    
+    static func <> (lhs: Reducer, rhs: Reducer) -> Reducer {
+        return Reducer { s, a in
+            rhs.reduce(&s, a)
+            lhs.reduce(&s, a)
+        }
+    }
+    
+    let reduce: (inout S, A) -> ()
+}
+
+private enum GraphMask {
+    case basis(WeightedGraph<Pitch.Class, Double>)
+    case adjacency(Graph<Int>)
+    case scaling(WeightedGraph<Int, Double>)
+}
+
+struct Getter<A, B> {
+    let view: (A) -> B
+}
+
+extension Getter {
+    init(_ keyPath: KeyPath<A, B>) {
+        self.init { $0[keyPath: keyPath] }
+    }
+}
+
+struct PullBack<A, B, C, P: SymmetricPair & Hashable, Q: SymmetricPair & Hashable> where P.A == A, Q.B == B {
```

Comment:
🙌.

You might be able to shove `P: SymmetricPair & Hashable` in the `where` clause.

```Swift
struct PullBack <A,B,C,P,Q> where 
    P: SymmetricPair & Hashable, 
    Q: SymmetricPair & Hashable,
    P.A == A,
    Q.B == B
{ ... }
```

Similar to the `Getter<?,?>` thought, what if these generic parameters had names. I thought for a second what they would be, but maybe you'd have a cleaner line of thinking :) !

Reply:
This shoving worked! Will be pushed (shoved) shortly.